Sources/SwiftInspectorVisitors/TypeSyntax+NameExtractor.swift

```diff
@@ -0,0 +1,25 @@
+// Created by dan_federman on 1/30/21.
+// Copyright © 2021 Airbnb Inc. All rights reserved.
+
+import SwiftSyntax
+
+extension TypeSyntax {
+
+  /// Returns the the qualified name for the type the receiver represents.
+  var qualifiedName: String {
+    if let typeIdentifier = self.as(SimpleTypeIdentifierSyntax.self) {
+      return typeIdentifier.name.text
+
+    } else if let typeIdentifier = self.as(MemberTypeIdentifierSyntax.self) {
+      let baseName = typeIdentifier.baseType.qualifiedName
+      return "\(baseName).\(typeIdentifier.name.text)"
+
+    } else {
```

Comment:
I don't know how to test this `else` case since it's explicitly covering a case that isn't expected. Even if I were convinced that both `SimpleTypeIdentifierSyntax` and `MemberTypeIdentifierSyntax` were the only types that we could encounter, there's no way for me to avoid an `else` without a force unwrap somewhere.

Given that I don't know how to test this condition, I think we're likely going to need to merge through the code coverage reduction. Open to ideas on how to avoid doing that.

Reply:
@dfed I think we could test this by manually constructing a `TypeSyntax` object with an invalid `raw.kind` value. If the value is anything other than `.simpleTypeIdentifier` or `.memberTypeIdentifier` we will end up in the `else`.

https://github.com/apple/swift-syntax/blob/577d3bbe3a4f39ba8843093f3bd1ae0b593da557/Sources/SwiftSyntax/gyb_generated/syntax_nodes/SyntaxTypeNodes.swift#L68

https://github.com/apple/swift-syntax/blob/577d3bbe3a4f39ba8843093f3bd1ae0b593da557/Sources/SwiftSyntax/gyb_generated/syntax_nodes/SyntaxTypeNodes.swift#L175

Reply:
The problem will then run into is that we're going to assert, but we could fix that too by having an internal function we call into from here.

```swift
extension TypeSyntax {

  var qualifiedName: String { computeQualifiedName(assertionFailure) }

  func computeQualifiedName(_ failureClosure: (String) -> Void) { /* ... */ }
}
```

Reply:
Hadn't occurred to me that we'd be able to create an unexpected object that conformed to `TypeSyntax`. Managed to get it done in 744be8c5e6e25d7600504250cac3d8560b6c82bd